RecoLocalTracker/Phase2TrackerRecHits/test/RecHitsValidation.cc

```diff
@@ -264,13 +487,37 @@ std::map< unsigned int, RecHitHistos >::iterator Phase2TrackerRecHitsValidation:
     return insertedIt.first;
 }
 
+
 unsigned int Phase2TrackerRecHitsValidation::getLayerNumber(const DetId& detid, const TrackerTopology* topo) {
     if (detid.det() == DetId::Tracker) {
-        if (detid.subdetId() == PixelSubdetector::PixelBarrel) return (topo->pxbLayer(detid));
-        else if (detid.subdetId() == PixelSubdetector::PixelEndcap) return (100 * topo->pxfSide(detid) + topo->pxfDisk(detid));
-        else return 999;
+        if (detid.subdetId() == PixelSubdetector::PixelBarrel) {
+          return (topo->pxbLayer(detid));
+        } else if (detid.subdetId() == PixelSubdetector::PixelEndcap) {
+          return (100 * topo->side(detid) + topo->pxfDisk(detid));
+        } else if (detid.subdetId() == StripSubdetector::TIB) {
+          return (topo->layer(detid));
+        } else if (detid.subdetId() == StripSubdetector::TOB) {
+          return (topo->layer(detid));
+        } else if (detid.subdetId() == StripSubdetector::TID) {
+          return (100 + topo->tidRing(detid));
+        } else if (detid.subdetId() == StripSubdetector::TEC) {
+          return (100 + topo->tecRing(detid));
+        } else {
+	  return 999;
+	}
```

Comment:
In Phase2, such as in Run2 and Phase1, the Pixel/StripSubdetectors name are still the same. In this moment, for Phase2 just 4 of them are used: PixelBarrel, PixelEndcap, TOB, TID.
@lowette Regarding specifically this part of the code, I would suggest to use just the function `layer()` which should return the correct number of the layer/disk. You can check it out [here](
https://github.com/lowette/cmssw/blob/11f38d18890775aea173fe11ed34025c568abff0/DataFormats/TrackerCommon/src/TrackerTopology.cc#L50).